app/controllers/allocations_controller.rb

```diff
@@ -18,19 +22,16 @@ def new
     @unavailable_pom_count = unavailable_pom_count
   end
 
-  # rubocop:disable Metrics/LineLength
   def show
     @prisoner = offender(nomis_offender_id_from_url)
-    primary_pom_nomis_id = Allocation.primary_pom_nomis_id(@prisoner.offender_no)
+    primary_pom_nomis_id = AllocationVersion.find_by(nomis_offender_id: @prisoner.offender_no).primary_pom_nomis_id
     @pom = PrisonOffenderManagerService.get_pom(active_caseload, primary_pom_nomis_id)
     @keyworker = Nomis::Keyworker::KeyworkerApi.get_keyworker(active_caseload, @prisoner.offender_no)
     @history = AllocationService.offender_allocation_history(@prisoner.offender_no)
   end
-  # rubocop:enable Metrics/LineLength
 
-  # rubocop:disable Metrics/MethodLength
   def edit
-    unless AllocationService.active_allocation?(nomis_offender_id_from_url)
+    unless AllocationVersion.find_by(nomis_offender_id: nomis_offender_id_from_url)
```

Comment:
Does this need constraining to active allocations?

Reply:
It's possible that an Allocation may have no primary pom assigned so I assume we just need to check for the existence of a record else they need to be redirected to create on

Reply:
Yeah, I think no allocated_poms (primary or secondary) is probably a fair proxy for inactive-allocation

Reply:
Done! We now check whether an allocation is active